pkg/scylla/api/cqlclient/v1alpha1/types_connectionconfig.go

```diff
@@ -0,0 +1,124 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Datacenter struct {
+	// CertificateAuthorityPath is the path to a cert file for the certificate authority.
+	// +optional
+	CertificateAuthorityPath string `json:"certificateAuthorityPath,omitempty"`
+
+	// CertificateAuthorityData contains PEM-encoded certificate authority certificates. Overrides CertificateAuthorityPath.
+	// +optional
+	CertificateAuthorityData []byte `json:"certificateAuthorityData,omitempty"`
+
+	// Server is the address of the Scylla cluster. All requests are sent to this address.
+	// Example: https://hostname:port
+	Server string `json:"server"`
+
+	// nodeDomain is the domain suffix that is concatenated with the host_id of the node driver wants to connect to.
+	// The concatenated string is then set as TLS server name for this TLS connection. This value is only used
+	// for setting the ServerName in TLS ClientHello, never resolved or used as a server address.
+	// Example: host_id.<nodeDomain>
+	// Client uses this nodeDomain as bare TLS server name when reaching out to the load balanced (discovery) endpoint.
+	ServiceDomain string `json:"nodeDomain"`
```

Comment:
I like `NodeDomain` better and it's in sync with json representation.  It's in fact domain pointing to scylla nodes, either any or particular one via subdomain.

Reply:
I am not too happy about that name as it seems only like a subset of what it is used for:
- `TLS.ServerName :uuid-1.cql.sc-name.sc-namespace` (`cql.sc-name.sc-namespace` is the node domain)
- `TLS:ServerName: cql.sc-name.sc-namespace` (`cql.sc-name.sc-namespace` doesn't seem like a domain used for nodes but rather a service DNS name) - it used to be different when we had `any.` subdomain there.
but I'll keep my reasons for v1alpha1 and use the wire name here.